finality-aleph/src/new_network/mod.rs

```diff
@@ -1,16 +1,22 @@
 // Everything here is dead code, but I don't want to create one enormous PR.
 #![allow(dead_code)]
-use codec::{Decode, Encode};
+use aleph_bft::Recipient;
+use codec::{Codec, Decode, Encode};
 use futures::stream::Stream;
 use sc_network::{Event, Multiaddr, NotificationSender, PeerId as ScPeerId};
```

Comment:
Do we really want this new module to be dependent on substrate?

Reply:
Maciek is working on making this dependence slightly weaker (or rather this will be a side result of what he is working on).

Unfortunately at least `PeerId`s and `Multiaddr`s are necessary, since a big part of the logic here deals with keeping connections alive. These could technically be type parameters for the whole module, but the template abundance that would result from that could rival C++, not in a good way.

Eh, now that you mentioned it I _could_ be tempted to make this more generic...